hw/mcu/ambiq/apollo3/include/mcu/hal_apollo3.h

```diff
@@ -60,6 +60,11 @@ struct apollo3_timer_cfg {
     uint8_t source;
 };
 
+void apollo3_periph_create(void);
+
+int apollo3_spi_set_ss_pin(int spi_num, int8_t ss_pin);
```

Comment:
These would be better off just being static functions inside `spi_apollo3.c`.

Reply:
I wanted to provide these functions to any BSPs that don't used the BUS_DRIVER.  Unfortunately MyNewt doesn't supply a HAL API for these functions.